app/Http/Controllers/CampController.php

```diff
@@ -757,13 +758,14 @@ public function campSubscription(Request $request, Validate $validate)
                 $msg = trans('message.success.unsubscribed');
             }
             $campSubscription->save();
-            $subscriptionId = ($subscription_id) ? "" : $campSubscription->id;
+            $subscriptionId = ($subscription_id && !$all['checked']) ? "" : $campSubscription->id;
             $response = new stdClass();
             $response->msg = $msg;
             $response->subscriptionId = $subscriptionId;
             $indexes = ['msg', 'subscriptionId'];
             $data[0] = $response;
             $data = $this->resourceProvider->jsonResponse($indexes, $data);
+            $data=$data[0];
             return $this->resProvider->apiJsonResponse(200, trans('message.success.success'), $data, '');
         } catch (Exception $e) {
             return $this->resProvider->apiJsonResponse(400, trans('message.error.exception'), $e->getMessage(), '');
```

Comment:
I think `$e->getMessage() ` should be the last not the third parameter

Reply:
yes. you are right